eventdiscovery-sdk/src/main/java/com/schedjoules/eventdiscovery/eventlist/itemsprovider/EventPageQueries.java

```diff
@@ -18,60 +18,75 @@
 package com.schedjoules.eventdiscovery.eventlist.itemsprovider;
 
 import com.schedjoules.client.ApiQuery;
+import com.schedjoules.client.eventsdiscovery.Envelope;
+import com.schedjoules.client.eventsdiscovery.Event;
 import com.schedjoules.client.eventsdiscovery.ResultPage;
 
+import java.util.HashMap;
 import java.util.Map;
 
 
 /**
- * Convenience, scrolling direction aware wrapper around {@link ResultPage} to get the next page query.
+ * Implementation for {@link PageQueries}.
  *
  * @author Gabor Keszthelyi
  */
-public final class DirectionalQuery<E>
+public final class EventPageQueries implements PageQueries<Envelope<Event>>
 {
-    private final ScrollDirection mDirection;
-    private final Map<ScrollDirection, ResultPage<E>> mResultPages;
+    private final Map<ScrollDirection, ResultPage<Envelope<Event>>> mLastResultPages = new HashMap<>(2);
 
 
-    public DirectionalQuery(Map<ScrollDirection, ResultPage<E>> resultPages, ScrollDirection direction)
+    @Override
+    public void saveLastResult(ResultPage<Envelope<Event>> resultPage, ScrollDirection direction)
     {
-        mResultPages = resultPages;
-        mDirection = direction;
+        mLastResultPages.put(direction, resultPage);
     }
 
 
-    public boolean hasQuery()
+    @Override
+    public boolean hasComingPage(ScrollDirection direction)
     {
-        ResultPage<E> resultPage = mResultPages.get(mDirection);
+        ResultPage<Envelope<Event>> resultPage = mLastResultPages.get(direction);
         if (resultPage == null)
         {
             return false;
         }
 
-        switch (mDirection)
+        switch (direction)
```

Comment:
IMO doing a Switch with enum is bad style in most cases.
Why not let the enum handle this like so?
```
   return direction.isAvailable(mLastResultPages.get(direction));
```
No Switch, no (stale) Exception.